### ISSUE TITLE
Editing a new theme affects the Godot's theme

### ISSUE THREAD

**Issue by viktorias** [godotengine/godot#4587]
Operating system or device - Godot version: Xubuntu 14.04.4 - Godot 2.0.2
Issue description (what happened, and what was expected):
So I was creating a new theme for a game I have been working on. And I have redefined the style of the Window dialogue, the LineEdit and the Label. So here is how the Godot engine is looking now:
![theme_bug](https://cloud.githubusercontent.com/assets/2116246/15099194/4b7b34fe-154f-11e6-8283-d3eab58226a5.png)
The Label did no get affected (in the theme the font color is black and on the screen it stays white), but the dialogue window and the line edits background are affected. But for some strange reason the font colors are not affected, and in this case the engines dialogues become hard to read.

I was expecting the engine's theme to be 100% independent of my game's theme.

Steps to reproduce:
Create a scene, add a window dialogue as root node
Theme property -> New Theme -> Edit
Theme menu -> Add class item -> Select Window Dialogue  and click add all 
Panel -> edit add a new image texture
Save the scene and enjoy the new look cause the Save scene screen has changed too...

Link to minimal example project (optional but very welcome):
See the screenshot.

**Comment by CowThing** [comment]
I have been able to reproduce this too. It only happens when you edit the default resource for the theme's item instead of creating a new resource for whatever item is being edited.

To fix this you can close Godot and reopen it, the engine theme should go back to normal, but the game will still have the proper theme.

**Comment by bojidar-bg** [comment]
@CowThing A better workaround is to create new StyleBox instead of editing the existing one :smile:

**Comment by viktorias** [comment]
@bojidar-bg Yes, just one has to make sure that he creates the new style first. Because if you edit the resource first, then the editor theme changes. And if you create a new style after then, that will not restore the original theme of the editor. Just reopening Godot or going back to the project list and back to the project will.
